lib/mongoid/traversable.rb

```diff
@@ -27,7 +27,7 @@ def _parent=(p)
     #
     # @return [ Array<Document> ] All child documents in the hierarchy.
     def _children
-      @__children ||= collect_children
```

Comment:
The memoization is what is causing the problems, though. Did you read the issue at https://github.com/mongoid/mongoid/issues/3991? There are myriad situations where the memoization can screw you, and the downside is horrible: Mongoid won't generate an UPDATE command and will silently fail to save documents, while returning `true` to the `save` command.